spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,990 @@
+      end
+      expect(subnets).to be_nil
+    end
+
+    it "should return valid non empty response for vms" do
+      vms = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/vms', :record => :new_episodes) do
+        vms = @vsd_client.get_vms
+      end
+      assert_object_not_empty(vms)
+      assert_vms(vms)
+    end
+
+    it "should return empty response for vms" do
+      vms = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/vms_empty', :record => :new_episodes) do
+        vms = @vsd_client.get_vms
+      end
+      expect(vms).to be_nil
+    end
+
+    it "should return valid non empty response for policy_groups" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups
+      end
+      assert_object_not_empty(policy_groups)
+      assert_policy_groups(policy_groups)
+    end
+
+    it "should return valid non empty response for policy_group by id" do
+      policy_group = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_group', :record => :new_episodes) do
+        policy_group = @vsd_client.get_policy_group(POLICY_GROUP_ID)
+      end
+
+      assert_object_not_empty(policy_group)
+      assert_policy_group(policy_group)
+    end
+
+    it "should return valid non empty response for policy_groups for domain" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_for_domain', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups_for_domain(DOMAIN_ID)
+      end
+
+      assert_object_not_empty(policy_groups)
+      assert_policy_groups(policy_groups)
+    end
+
+    it "should return empty response for policy_groups" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_empty', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups
+      end
+      expect(policy_groups).to be_nil
+    end
+
+    it "should return empty response for policy_group by id" do
+      policy_group = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_group_empty', :record => :new_episodes) do
+        policy_group = @vsd_client.get_policy_group(POLICY_GROUP_ID)
+      end
+      expect(policy_group).to be_nil
+    end
+
+    it "should return empty response for policy_groups for domain" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_for_domain_empty', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups_for_domain(DOMAIN_ID)
+      end
+      expect(policy_groups).to be_nil
+    end
+  end
+
+  it "should fail on wrong password" do
```

Comment:
Not really neccessary, but I'd wrap these last three `it`-s into a
```ruby
context "when login fails" do ...
```
In RubyMine we will then be able to collapse them when needed 😄